File: frontend/src/api/apiResponse.ts
export interface PageMeta {
  page: number
  size: number
  totalElements: number
  totalPages: number
}

export interface ApiResponse<T> {
  success: boolean
  message: string
  data: T
  meta?: PageMeta
}

export interface ApiErrorItem {
  field: string
  message: string
}

export interface ApiErrorResponse {
  success: false
  message: string
  errors?: ApiErrorItem[]
}

File: backend/src/main/java/com/englishcenter/common/api/ApiResponse.java
package com.englishcenter.common.api;

import com.fasterxml.jackson.annotation.JsonInclude;

@JsonInclude(JsonInclude.Include.NON_NULL)
public record ApiResponse<T>(
        boolean success,
        String message,
        T data,
        PageMeta meta
) {
    public static <T> ApiResponse<T> success(T data) {
        return new ApiResponse<>(true, "Success", data, null);
    }

    public static <T> ApiResponse<T> success(T data, PageMeta meta) {
        return new ApiResponse<>(true, "Success", data, meta);
    }

    public static <T> ApiResponse<T> error(String message) {
        return new ApiResponse<>(false, message, null, null);
    }
}
